FreeRTOS/Lab-Project-FreeRTOS-FAT
# Change the FF_MakeNameCompliant behavior

* Change the handling of invalid file-names to cause an error. Depending on the entry to create (directory/file), a different error code is returned.

Description
-----------
This changes the behavior of the FF_CreateDirent function to return an error in case of an invalid path. 
Filenames with the first character of 0xE5 are still modified. (Which might have some issues. TODO)

Note: Invalid directory paths and invalid file paths now result in two different FreeRTOS+ return values. Is this the intended behavior or should only one error code be used?
```
case FF_ERR_FILE_INVALID_PATH:
    return pdFREERTOS_ERRNO_ENOTDIR; /* The path of the file was not found. */
case FF_ERR_DIR_INVALID_PATH:
    return pdFREERTOS_ERRNO_EINVAL; /* Could not find the directory specified by the path. */
```

FF_MakeNameCompliant returns a boolean and not an error-code, because the function
would need an additional parameter to select the right error code (FF_ERR_DIR_INVALID_PATH or FF_ERR_FILE_INVALID_PATH).

Test Steps
-----------
A invalid filepath "test:file.txt" resulted in setting the FF_ERR_FILE_INVALID_PATH. (Tested through breakpoint in prvFFErrorToErrno).

Todo:
- mkdir with invalid path
- handling of names starting with 0xE5

Checklist:
----------



Related Issue
-----------


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
